core/src/main/java/org/geysermc/geyser/translator/collision/GlassPaneCollision.java

```diff
@@ -0,0 +1,111 @@
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 2;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=false")) {
+            facing = 3;
+        } else if (params.contains("west=true") && params.contains("north=false") && params.contains("east=false") && params.contains("south=false")) {
+            facing = 4;
+        }
+        if (params.contains("north=true") && params.contains("east=true") && params.contains("south=false") && params.contains("west=false")) {
+            facingB = 1;
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=true") && params.contains("west=false")) {
+            facingB = 2;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=true")) {
+            facingB = 3;
+        } else if (params.contains("west=true") && params.contains("north=true") && params.contains("east=false") && params.contains("south=false")) {
+            facingB = 4;
+        }
+    }
+
+    @Override
+    public boolean correctPosition(GeyserSession session, int x, int y, int z, BoundingBox playerCollision) {
+        boolean result = super.correctPosition(session, x, y, z, playerCollision);
+        // Hack to prevent false positives
+        playerCollision.setSizeX(playerCollision.getSizeX() - 0.0001);
+        playerCollision.setSizeY(playerCollision.getSizeY() - 0.0001);
+        playerCollision.setSizeZ(playerCollision.getSizeZ() - 0.0001);
+
+        // Check for glass_pane bug (glass_pane are 0.5625 wide thick on Java but 0.5 blocks wide on Bedrock)
+        if (this.checkIntersection(x, y, z, playerCollision)) {
+            switch (facing) {
+                case 1 -> playerCollision.setMiddleZ(z + 0.8625); // North
+                case 2 -> playerCollision.setMiddleX(x + 0.1375); // East
+                case 3 -> playerCollision.setMiddleZ(z + 0.1375); // South
+                case 4 -> playerCollision.setMiddleX(x + 0.8625); // West
+            }
+            switch (facingB) {
+                case 1 -> playerCollision.setMiddleX(x + 0.1375); // East
+                case 2 -> playerCollision.setMiddleZ(z + 0.1375); // South
+                case 3 -> playerCollision.setMiddleX(x + 0.8625); // West
+                case 4 -> playerCollision.setMiddleZ(z + 0.8625); // North
+            }
+        }
+
+        playerCollision.setSizeX(playerCollision.getSizeX() + 0.0001);
+        playerCollision.setSizeY(playerCollision.getSizeY() + 0.0001);
+        playerCollision.setSizeZ(playerCollision.getSizeZ() + 0.0001);
+        return result;
+    }
+
+
+    public boolean checkIntersection(double x, double y, double z, BoundingBox playerCollision) {
```

Comment:
This method is not needed, as it is identical to the `checkIntersection()` method of the super class